text/0016-auto-code-mappings.md

```diff
@@ -0,0 +1,86 @@
+
+Alternatively, we could analyze stack traces when new issues are generated when there are no code mappings for a project. I believe this approach would be wasteful.
+
+NOTES:
+
+* POC for JS still need to be completed
+* POC testing of a different API approach than the Search API is still needed
+  * Rated at 30 requests per minute; not scalable
+  * Only default branch; limiting our potential options
+  * Inaccurate at times (as per @asottile-sentry)
+
+### Considerations
+
+* Not all projects have a platform set (e.g. other), thus, we will need to look at file extensions
+* There are some orgs that have more than one Github org associated to their Sentry org
+  * I don't think this is a problem
+* There are some projects that have code from more than one repository
+* Some projects have more than one code mapping per repository for a given project
+  * For instance, Sentry has `sentry` and `sentry_plugins` code mappings
+* In the stack traces we will have frame for projects that are 3rd party libraries (e.g. `requests`)
+  * If the org vendors the package in at least one repo we may end up creating a code mapping to it
+    * Potential for a bug since it may be pointing to an old vendored version
+    * Issues come with info about packages. I wonder if in some cases we may be able to detect this.
+* Not all orgs will have granted us access to All their Github repositories upon installation (or modified it later)
+  * This will reduce the number of exact matches
+* As part of the automation we will be creating Repository objects
+  * If we end up adding a lot of repos for the org we may be indirectly causing UI issues when there’s no proper pagination including dropdowns
+* Available APIs
+  * Search API (proven to work in POC)
+    * The [Search API](https://docs.github.com/en/rest/search#rate-limit) has a rate limit of 30 requests per minute, thus, at best we could create 30 code mappings per minute
+      * **UNKNOWN** We may be able to create multiple tokens to increase the capacity 
+  * [Repositories](https://docs.github.com/en/rest/repos/repos#list-organization-repositories) + [Trees API](https://docs.github.com/en/rest/git/trees#get-a-tree)
+    * To be investigated in current sprint (**UNKNOWN**)
+    * Using the `recursive` parameter returns all objects in the repo (almost 12k)
+* Code map derivation has not yet been tested against JS stack traces (**UNKNOWN**)
+  * Scheduled to be tackled in current sprint
+
+## Drawbacks
+
+None I can think of.
+
+## Unresolved questions
+
+* We are leaning toward not notifying users that code mappings were added to a project that did not have code mappings since they didn't even know there were missing out in any features. In the future, in cases where code mappings are already in place for a project, we will need to re-evaluate and discuss it since we may cause workflow regressions for ownership rules.
```

Comment:
I think the nitty gritty isn't that important to convey but not communicating that this change has been made might introduce some user confusion. My opinion is that changes to user-editable details (even minor) should probably be surfaced in some way.

Plus, if we're making those updates messaged, it'd also be a venue to add it this feature to the notification platform. If you restructure your app (e.g. `source/` to `src/`), features in Sentry will break without telling you they need attention. Adding these notifications for managers/owners might grant some visibility onto the features, and we can just leverage that for automatic updates like this doc specs out.

Reply:
This pass will not focus on editing values but adding code mappings where there are none set.